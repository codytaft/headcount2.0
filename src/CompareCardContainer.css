.compareCardContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: auto;
  justify-content: center;
  margin: 1rem 1rem 0 1rem;
}