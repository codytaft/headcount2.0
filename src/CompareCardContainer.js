import React from 'react'
import { CompareCard } from './CompareCard'
import './CompareCardContainer.css'

export const CompareCardContainer = ({ comparedObject }) => {
  if (comparedObject) {
    const compareKeys = Object.keys(comparedObject)
    var compareCards =
      <CompareCard
        locationA={compareKeys[0]}
        locationAvgA={comparedObject[compareKeys[0]]}
        locationB={compareKeys[1]}
        locationAvgB={comparedObject[compareKeys[1]]}
        comparison={comparedObject.compared}
      />
  }
  
  return (
    <div className="compareCardContainer">
      { compareCards }
    </div>
  )
}