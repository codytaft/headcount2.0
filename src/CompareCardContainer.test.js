import React from 'react';
import { shallow } from 'enzyme';
import { CompareCardContainer } from './CompareCardContainer.js'

describe('CompareCardContainer', () => {

  it('should match the snapshot with no cards', () => {
    const mockProp = {
                      comparedObject: {}
    }
    const wrapper = shallow(<CompareCardContainer comparedObject={mockProp}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with cards', () => {
    const mockProp = {
                      comparedObject: {
                        'ACADEMY 20': 0.407,
                        'ADAMS COUNTY 14': 0.709,
                        'compared': 1.742
                      }
    }
    const wrapper = shallow(<CompareCardContainer comparedObject={mockProp}/>)
    expect(wrapper).toMatchSnapshot()
  })

})