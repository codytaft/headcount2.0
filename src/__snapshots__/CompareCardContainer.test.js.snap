// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCardContainer should match the snapshot with cards 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    comparedObject={
        Object {
            "comparedObject": Object {
              "ACADEMY 20": 0.407,
              "ADAMS COUNTY 14": 0.709,
              "compared": 1.742,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        comparison={undefined}
        locationA="comparedObject"
        locationAvgA={
                Object {
                        "ACADEMY 20": 0.407,
                        "ADAMS COUNTY 14": 0.709,
                        "compared": 1.742,
                      }
        }
        locationAvgB={undefined}
        locationB={undefined}
/>,
      "className": "compareCardContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "comparison": undefined,
        "locationA": "comparedObject",
        "locationAvgA": Object {
          "ACADEMY 20": 0.407,
          "ADAMS COUNTY 14": 0.709,
          "compared": 1.742,
        },
        "locationAvgB": undefined,
        "locationB": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          comparison={undefined}
          locationA="comparedObject"
          locationAvgA={
                    Object {
                              "ACADEMY 20": 0.407,
                              "ADAMS COUNTY 14": 0.709,
                              "compared": 1.742,
                            }
          }
          locationAvgB={undefined}
          locationB={undefined}
/>,
        "className": "compareCardContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparison": undefined,
          "locationA": "comparedObject",
          "locationAvgA": Object {
            "ACADEMY 20": 0.407,
            "ADAMS COUNTY 14": 0.709,
            "compared": 1.742,
          },
          "locationAvgB": undefined,
          "locationB": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CompareCardContainer should match the snapshot with no cards 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    comparedObject={
        Object {
            "comparedObject": Object {},
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        comparison={undefined}
        locationA="comparedObject"
        locationAvgA={Object {}}
        locationAvgB={undefined}
        locationB={undefined}
/>,
      "className": "compareCardContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "comparison": undefined,
        "locationA": "comparedObject",
        "locationAvgA": Object {},
        "locationAvgB": undefined,
        "locationB": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          comparison={undefined}
          locationA="comparedObject"
          locationAvgA={Object {}}
          locationAvgB={undefined}
          locationB={undefined}
/>,
        "className": "compareCardContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparison": undefined,
          "locationA": "comparedObject",
          "locationAvgA": Object {},
          "locationAvgB": undefined,
          "locationB": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
